### PR TITLE
Expanded support for musl/linux syscalls (sysinfo, getrusage, prlimit64, umask, fchmodat)

### PR DESCRIPTION
**Pull request by FlareCoding**
> [!NOTE]
> **Low Risk**
> Additive syscall and per-process metadata with stub chmod/umask behavior; prlimit limits are stored but not newly enforced on open/stack paths in this diff.
> 
> **Overview**
> Adds several **Linux/musl-facing syscalls** and wires them into the aarch64/x86_64 syscall number tables and `syscall_table.cpp`.
> 
> **Process state:** Each `thread_group` now holds per-process **resource limits** (`rlimits`) and a **umask**. New processes inherit both from the creator (POSIX-style) or get defaults (`init_default_rlimits`, umask `022`). **`prlimit64`** reads/writes limits for the caller’s PID only (with validation); **`umask`** atomically swaps the mask under the group lock.
> 
> **Reporting syscalls:** **`sysinfo`** fills uptime, RAM totals/free from the PMM, and a live process count. **`getrusage`** returns mostly-zero `rusage` except CPU time from `run_ticks` (all counted as user time; children are zero).
> 
> **Permission stubs:** **`fchmodat`** / x86 **`chmod`** resolve the path like other `*at` handlers and succeed without changing modes (VFS has no stored mode bits). Same pattern as existing `faccessat` stubs.
> 
> Kernel unit tests cover EINVAL/EFAULT/ESRCH edge cases; fuller behavior is left to userland.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 27acf543eee58c4b39b6626e46a2a057740f2db5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: kernel/syscall/handlers/sys_fd.cpp [hunk 1]
```diff
@@ -1481,6 +1481,56 @@ DEFINE_SYSCALL3(faccessat, dirfd, pathname, mode) {
     return 0;
 }
 
+DEFINE_SYSCALL4(fchmodat, dirfd, pathname, mode, flags) {
+    // AT_SYMLINK_NOFOLLOW is the only defined flag
+    if (flags & ~0x100ULL) {
+        return syscall::EINVAL;
+    }
```
[the remark this answers is not in the document]
Comment (FlareCoding) [reply kernel/syscall/handlers/sys_fd.cpp]: fixed

File: kernel/syscall/handlers/sys_fd.cpp [hunk 2]
```diff
@@ -1481,6 +1481,56 @@ DEFINE_SYSCALL3(faccessat, dirfd, pathname, mode) {
...
+
+    if (mode & ~07777ULL) {
+        return syscall::EINVAL;
+    }
```
[the remark this answers is not in the document]
Comment (FlareCoding) [reply kernel/syscall/handlers/sys_fd.cpp]: fixed
